TheAlienThatEatsTheCarrot/GameEngine/Systems/GameSystems/PhysicsSystem.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  PhysicsSystem.swift
+//  TheAlienThatEatsTheCarrot
+//
+//  Created by Sun Xinyu on 2/4/24.
+//
+
+import CoreGraphics
+
+final class PhysicsSystem: System {
+    let nexus: Nexus
+    let physicsWorld: PhysicsWorld
+
+    init(nexus: Nexus, physicsWorld: PhysicsWorld) {
+        self.nexus = nexus
+        self.physicsWorld = physicsWorld
+    }
+
+    func update(deltaTime: CGFloat) {
+        let entities = nexus.getEntities(with: PhysicsComponent.self)
+
+        entities.forEach { entity in
+            updateRenderable(entity)
+        }
```

Comment:
not sure about this but if physics body is updated after renderable, wouldn't the positions of renderables be out of sync with the positions of physics body?

Would it make more sense if we update renderables after physics bodies?